src/main/java/com/direwolf20/buildinggadgets/common/util/blocks/RegionSnapshot.java

```diff
@@ -82,32 +78,29 @@ private static NBTTagCompound serialize(NBTTagCompound tag, RegionSnapshot snaps
             }
             // Palette serialization ends
 
-            // Hence we record the storage frame after it is completed, the initial values of the counter would be used as a storage frame regardless, therefore we need to skip inorder for this algorithm to function correctly
-            if (firstIteration)
-                firstIteration = false;
-            else {
-                // If the same block state showed up again, increase the number of streaks
-                // If we have more than 255 streaks (maximum integer that can be stored in 8 bits), we force start a new streak, because the current storage frame won't fit that many of them
-                if (state.equals(lastStreak) && streak < 255) {
-                    streak++;
-                    continue;
-                }
+            // If the same block state showed up again, increase the number of streaks
+            // If we have more than 255 streaks (maximum integer that can be stored in 8 bits), we force start a new streak, because the current storage frame won't fit that many of them
+            if (state.equals(lastStreak) && streak < 255) {
+                streak++;
+                continue;
+            }
 
-                // Note: if the program reached here, it means the current storage frame is completed
+            // Note: if the program reached here, it means the current storage frame is completed
 
-                // Number of streaks and the ID of the block state that is streaking
-                // - no need to differentiate between single and repeating block states, since the number at the larger endian represents how many repeating block states are there
-                // - if the streaking block state is empty (returns `null` when `get()` is called), it will be mapped to ID 0.
-                //   - see initialization of `mapPalettes`
-                frames.add(((streak & 0xff) << 24) | (mapPalettes.get(lastStreak.get()) & 0xffffff));
-            }
+            // Number of streaks and the ID of the block state that is streaking
+            // - no need to differentiate between single and repeating block states, since the number at the larger endian represents how many repeating block states are there
+            // - if the streaking block state is empty (returns `null` when `get()` is called), it will be mapped to ID 0.
```

Comment:
According to the [docs](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get--, the optional **throws an NoSuchElementException** when it is empty and `get` is called. Thid needs tobeither use `ifPresent` or check if it `isPresent` (presumably the latter, as you want to do something in both cases)

Reply:
I want just the raw form of it (`null` is fine), maybe I'll use `.orElse(null)` instead.
This is because `Map` says `null` will always be treated as the same key, but `Optional.empty()` does not enforce it always return the same value (i.e. when the streaking block state is `Optional.empty()`, it might not work according to the docs).